File: brew_wizard_ui/src/app/equipment/equipment.ts
export interface IEquipment {
  id: number;
  user_id: number;
  name: string;
  global: boolean;
  fly_sparge: boolean;
  batch_sparge: boolean;
  wl_hlt: number;
  wl_mash: number;
  wl_boil: number;
  boil_rate: number;
  efficiency: number;
  batch_limit: number;
  whirlpool: boolean;
  hop_back: boolean;
  recipe_count: number;
}

File: brew_wizard_server/app/models/equipment.rb
class Equipment < ApplicationRecord
  belongs_to :user

  validates :name, presence: true, length: { maximum: 40 }
  validates :wl_hlt, numericality: { greater_than_or_equal_to: 0, less_than: 100}
  validates :efficiency, numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: 100}
  validates :wl_mash, numericality: { greater_than_or_equal_to: 0, less_than: 100}
  validates :wl_boil, presence: true, numericality: { greater_than_or_equal_to: 0, less_than: 100}
  validates :boil_rate, presence: true, numericality: { greater_than_or_equal_to: 0, less_than: 100}
  validates :description, length: { maximum: 500 }
  validates :batch_limit, presence: true, numericality: { greater_than: 0 }

  def recipe_count
    return Recipe.where(equipment_id: self.id).count
  end
end
